Wrappers/Python/cil/optimisation/algorithms/FISTA.py

```diff
@@ -93,43 +101,56 @@ class ISTA(Algorithm):
     """
 
     def _provable_convergence_condition(self):
-        return self.step_size <= 0.99*2.0/self.f.L
+        if self.preconditioner is not None:
+            raise NotImplementedError("Can't check convergence criterion if a preconditioner is used ")
+
+        if isinstance(self.step_size_rule, ConstantStepSize): 
+            return self.step_size_rule.step_size <= 0.99*2.0/self.f.L
+        else:
+            raise TypeError("Can't check convergence criterion for non-constant step size")
 
     @property
     def step_size(self):
-       return self._step_size
+        if isinstance(self.step_size_rule, ConstantStepSize):
+            return self.step_size_rule.step_size
+        else:
+            raise TypeError("There is not a constant step size, it is set by a step-size rule")
```

Comment:
can't quite understand the English here, but wouldn't this fall back to calling the rule?